In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline


In [ ]:
season2018 = pd.read_csv("/content/drive/My Drive/MySaudiLeagueData2018.csv")
season2019 = pd.read_csv("/content/drive/My Drive/MySaudiLeagueData2019.csv")

In [ ]:
season1819.drop(columns=["home_coach", "away_coach"], inplace=True)
season1920.drop(columns=["home_coach", "away_coach"], inplace=True)

## initial data exploration


In [ ]:
season1819.head()

Unnamed: 0  ...     season
0           0  ...  2018/2019
1           1  ...  2018/2019
2           2  ...  2018/2019
3           3  ...  2018/2019
4           4  ...  2018/2019

[5 rows x 10 columns]

In [ ]:
season1819.shape

(240, 10)

We have 240 matches which makes snse, since we have 16 teams, and 30 round, so we got 16*30 / 2 = 240

In [ ]:
season1819.isnull().sum().sum()

0

In [ ]:
season1920.head()

Unnamed: 0  ...     season
0           0  ...  2019/2020
1           1  ...  2019/2020
2           2  ...  2019/2020
3           3  ...  2019/2020
4           4  ...  2019/2020

[5 rows x 10 columns]

In [ ]:
season1920.shape

(184, 10)

As of this moment 184 matches has been played which means that the dataset is complete

In [ ]:
season1920.isnull().sum().sum()

0

In [ ]:
season1819["home"].unique()

array(['Al Ettifaq', 'Al Hazem', 'Ohod Club', 'Qadisiyah', 'Al Hilal',
       'Al Shabab', 'Al Faisaly', 'Alahli', 'Al Fateh', 'Al Nassr',
       'Al Fayha', 'Al Raed', 'Al Ittihad', 'Al Taawoun', 'Al Wehda',
       'Al-Batin'], dtype=object)

In [ ]:
season1920["home"].unique()

array(['Al Nassr', 'Alahli', 'Al Hilal', 'Al Taawoun', 'Al Ittihad',
       'Al Fateh', 'Al Faisaly', 'Al Wehda', 'Al Adalh', 'Al Fayha',
       'Al Shabab', 'Al Ettifaq', 'Al Raed', 'Damac', 'Abha', 'Al Hazem'],
      dtype=object)

In [ ]:
len(season1819["home"].unique())

16

In [ ]:
len(season1920["home"].unique())

16

In [ ]:
season1819.columns

Index(['Unnamed: 0', 'home', 'away', 'home_goals', 'away_goals', 'home_format',
       'away_format', 'referee', 'players', 'season'],
      dtype='object')

In [ ]:
season1920.columns

Index(['Unnamed: 0', 'home', 'away', 'home_goals', 'away_goals', 'home_format',
       'away_format', 'referee', 'players', 'season'],
      dtype='object')

## Little discussion

I am not sure if this is the optimal format for the data, I am worried that since entries are split between Home & Away, when we want to perform some eda or predict the outcome we would have 2 target variables to work against. 
Another thing is that the machine would probably consider being home or away as significantly different when it is not.
Maybe we should change the format so that we have the following 
* [Team, Current points, IsHome, opponent, opponent current points, Date, Season, referee, Goals Scored]

Now the target variable is goals scored.
we can try to predict the final outcome of the match by running the model twice on both teams, then by comparing the goals we give the final result as Win, defeat, draw. 

## Date cleaning & processing, before getting back to EDA

In [ ]:
# before we do that we should join the two dataframes together
df = pd.concat([season1819, season1920])

In [ ]:
df.head()

Unnamed: 0  ...     season
0           0  ...  2018/2019
1           1  ...  2018/2019
2           2  ...  2018/2019
3           3  ...  2018/2019
4           4  ...  2018/2019

[5 rows x 10 columns]

In [ ]:
df.tail()

Unnamed: 0  ...     season
179         179  ...  2019/2020
180         180  ...  2019/2020
181         181  ...  2019/2020
182         182  ...  2019/2020
183         183  ...  2019/2020

[5 rows x 10 columns]

In [ ]:
df.shape

(424, 10)

In [ ]:
df.loc[0]

Unnamed: 0  ...     season
0           0  ...  2018/2019
0           0  ...  2019/2020

[2 rows x 10 columns]

In [ ]:
#the index seems to be mixed between the two seasons

In [ ]:
df.reset_index(inplace=True, drop=True)

In [ ]:
#let's get rid of the Unnamed: 0 variable
df.drop("Unnamed: 0", axis=1, inplace=True)

In [ ]:
df.columns

Index(['home', 'away', 'home_goals', 'away_goals', 'home_format',
       'away_format', 'referee', 'players', 'season'],
      dtype='object')

In [ ]:
df.head()

home  ...     season
0  Al Ettifaq  ...  2018/2019
1    Al Hazem  ...  2018/2019
2   Ohod Club  ...  2018/2019
3   Qadisiyah  ...  2018/2019
4    Al Hilal  ...  2018/2019

[5 rows x 9 columns]

##### Now we will change the format of the data


In [ ]:
# since every entry now is based on home I'll create a feature called IsHome and set it to 1

df["IsHome"] = 1

I will space out rows by changing the index to only even numbers, then inject the opposing team between the rows in the odd index

In [ ]:
len([x for x in range(0, len(df)*2, 2)])

424

In [ ]:
len(df)

424

In [ ]:
df.index = [x for x in range(0, len(df) * 2, 2)]

In [ ]:
df.columns

Index(['home', 'away', 'home_goals', 'away_goals', 'home_format',
       'away_format', 'referee', 'players', 'season', 'IsHome'],
      dtype='object')

In [ ]:
# I'll change the order of the columns so that it is close to the desired format

In [ ]:
df = df[["home", "home_format", "IsHome", "away", "away_format", "players","referee", "season", "home_goals", "away_goals"]]

In [ ]:
df.columns


Index(['home', 'home_format', 'IsHome', 'away', 'away_format', 'players',
       'referee', 'season', 'home_goals', 'away_goals'],
      dtype='object')

In [ ]:
df.head()

home home_format  IsHome  ...     season home_goals away_goals
0  Al Ettifaq     4-2-3-1       1  ...  2018/2019          1          1
2    Al Hazem     4-2-3-1       1  ...  2018/2019          0          0
4   Ohod Club     4-3-2-1       1  ...  2018/2019          1          2
6   Qadisiyah       4-3-3       1  ...  2018/2019          0          0
8    Al Hilal     4-1-3-2       1  ...  2018/2019          1          0

[5 rows x 10 columns]

In [ ]:
for i in [x for x in range(1, len(df) * 2 + 1, 2)]:
    old_entry = df.loc[i-1]
    df.loc[i] = [old_entry.away, old_entry.away_format, 0, old_entry.home, old_entry.home_format, old_entry.players, old_entry.referee, old_entry.season, old_entry.away_goals, np.nan]
    

In [ ]:
df.sort_index(inplace=True)

In [ ]:
df.head(10)

home home_format  IsHome  ...     season home_goals away_goals
0  Al Ettifaq     4-2-3-1       1  ...  2018/2019        1.0        1.0
1     Al Raed     4-2-3-1       0  ...  2018/2019        1.0        NaN
2    Al Hazem     4-2-3-1       1  ...  2018/2019        0.0        0.0
3    Al Wehda     4-2-3-1       0  ...  2018/2019        0.0        NaN
4   Ohod Club     4-3-2-1       1  ...  2018/2019        1.0        2.0
5    Al Nassr     4-3-2-1       0  ...  2018/2019        2.0        NaN
6   Qadisiyah       4-3-3       1  ...  2018/2019        0.0        0.0
7    Al Fateh     4-2-3-1       0  ...  2018/2019        0.0        NaN
8    Al Hilal     4-1-3-2       1  ...  2018/2019        1.0        0.0
9    Al Fayha       3-5-2       0  ...  2018/2019        0.0        NaN

[10 rows x 10 columns]

In [ ]:
# Now we will change the columns so that it aligns with our format

df.drop("away_goals", inplace=True, axis=1)

df.columns = ["Team1", "format", "IsHome", "Opponent", "Opponent_format", "Players", "Referee", "Season", "GoalsScored"]

In [ ]:
df.head(10)

Team1   format  ...     Season GoalsScored
0  Al Ettifaq  4-2-3-1  ...  2018/2019         1.0
1     Al Raed  4-2-3-1  ...  2018/2019         1.0
2    Al Hazem  4-2-3-1  ...  2018/2019         0.0
3    Al Wehda  4-2-3-1  ...  2018/2019         0.0
4   Ohod Club  4-3-2-1  ...  2018/2019         1.0
5    Al Nassr  4-3-2-1  ...  2018/2019         2.0
6   Qadisiyah    4-3-3  ...  2018/2019         0.0
7    Al Fateh  4-2-3-1  ...  2018/2019         0.0
8    Al Hilal  4-1-3-2  ...  2018/2019         1.0
9    Al Fayha    3-5-2  ...  2018/2019         0.0

[10 rows x 9 columns]

In [ ]:
df.GoalsScored = df.GoalsScored.astype("int64")

In [ ]:
df.head(2)

Team1   format  IsHome  ...         Referee     Season GoalsScored
0  Al Ettifaq  4-2-3-1       1  ...  Ummer Al-Senin  2018/2019           1
1     Al Raed  4-2-3-1       0  ...  Ummer Al-Senin  2018/2019           1

[2 rows x 9 columns]

#### Adding Current Points feature & final game state

In [ ]:
df.Team1.unique()

array(['Al Ettifaq', 'Al Raed', 'Al Hazem', 'Al Wehda', 'Ohod Club',
       'Al Nassr', 'Qadisiyah', 'Al Fateh', 'Al Hilal', 'Al Fayha',
       'Al Shabab', 'Al Ittihad', 'Al Faisaly', 'Al-Batin', 'Alahli',
       'Al Taawoun', 'Damac', 'Al Adalh', 'Abha'], dtype=object)

In [ ]:
df["pts"] = 0
df["opponent_pts"] = 0
df["result"] = 0

In [ ]:

for season in ["2018/2019", "2019/2020"]:
    
    season_df = df.loc[(df["Season"] == season )]

    points = {'Al Ettifaq':0,
              'Al Raed':0,
              'Al Hazem':0,
               'Al Wehda':0,
               'Ohod Club':0,
                'Al Nassr':0,
               'Qadisiyah':0,
               'Al Fateh':0,
               'Al Hilal':0,
               'Al Fayha':0,
                'Al Shabab':0,
                'Al Ittihad':0,
                'Al Faisaly':0,
                'Al-Batin':0,
                'Alahli':0,
               'Al Taawoun':0,
                'Damac':0,
                'Al Adalh':0,
                'Abha':0}

    for i in season_df.index:
        if i % 2 == 0 :
            Team1 = season_df.loc[i, "Team1"]
            Team2 = season_df.loc[i, "Opponent"]


            season_df.loc[i,"pts"] = points[Team1]
            season_df.loc[i, "opponent_pts"] = points[Team1]

            season_df.loc[i+1,"pts"] = points[Team2]
            season_df.loc[i+1, "opponent_pts"] = points[Team1]


            if season_df.loc[i, "GoalsScored"] > season_df.loc[i+1, "GoalsScored"]:
                points[Team1] += 3
                season_df.loc[i, "result"] = "Win"
                season_df.loc[i+1, "result"] = "Defeat"
            
            elif season_df.loc[i, "GoalsScored"] == season_df.loc[i+1, "GoalsScored"]:
                points[Team1] += 1
                points[Team2] += 1
                season_df.loc[i, "result"] = "Draw"
                season_df.loc[i+1, "result"] = "Draw"

            else:
                points[Team2] += 3
                season_df.loc[i, "result"] = "Defeat"
                season_df.loc[i+1, "result"] = "Win"

        df.loc[df["Season"] == season] = season_df




/usr/local/lib/python3.6/dist-packages/pandas/core/indexing.py:966: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self.obj[item] = s


In [ ]:
df.head(50)

Team1   format  IsHome  ... pts opponent_pts  result
0   Al Ettifaq  4-2-3-1       1  ...   0            0    Draw
1      Al Raed  4-2-3-1       0  ...   0            0    Draw
2     Al Hazem  4-2-3-1       1  ...   0            0    Draw
3     Al Wehda  4-2-3-1       0  ...   0            0    Draw
4    Ohod Club  4-3-2-1       1  ...   0            0  Defeat
5     Al Nassr  4-3-2-1       0  ...   0            0     Win
6    Qadisiyah    4-3-3       1  ...   0            0    Draw
7     Al Fateh  4-2-3-1       0  ...   0            0    Draw
8     Al Hilal  4-1-3-2       1  ...   0            0     Win
9     Al Fayha    3-5-2       0  ...   0            0  Defeat
10   Al Shabab  4-2-3-1       1  ...   0            0     Win
11  Al Ittihad  4-2-3-1       0  ...   0            0  Defeat
12  Al Faisaly  4-2-3-1       1  ...   0            0    Draw
13    Al-Batin    4-4-2       0  ...   0            0    Draw
14      Alahli  4-2-3-1       1  ...   0            0    Draw
15  Al Taawoun    5-3-2       0  ...   0            0    Draw
16  Al Ettifaq    4-4-2       1  ...   1            1     Win
17    Al-Batin    4-4-2       0  ...   1            1  Defeat
18    Al Fateh  4-2-3-1       1  ...   1            1    Draw
19    Al Wehda  4-2-3-1       0  ...   1            1    Draw
20    Al Hazem  4-2-3-1       1  ...   1            1    Draw
21  Al Taawoun  4-2-3-1       0  ...   1            1    Draw
22    Al Nassr  4-2-3-1       1  ...   3            3     Win
23  Al Faisaly  4-2-3-1       0  ...   1            3  Defeat
24    Al Fayha  4-2-3-1       1  ...   0            0  Defeat
25   Al Shabab  4-2-3-1       0  ...   3            0     Win
26   Ohod Club  4-2-3-1       1  ...   0            0  Defeat
27      Alahli  4-2-3-1       0  ...   1            0     Win
28     Al Raed    3-5-2       1  ...   1            1  Defeat
29    Al Hilal  4-2-3-1       0  ...   3            1     Win
30  Al Ittihad  4-2-3-1       1  ...   0            0  Defeat
31   Qadisiyah  4-2-3-1       0  ...   1            0     Win
32   Qadisiyah    4-4-2       1  ...   4            4  Defeat
33    Al Nassr  4-2-3-1       0  ...   6            4     Win
34   Al Shabab  4-2-3-1       1  ...   6            6    Draw
35    Al Fateh  4-2-3-1       0  ...   2            6    Draw
36  Al Taawoun  4-2-3-1       1  ...   2            2     Win
37  Al Ittihad  4-2-3-1       0  ...   0            2  Defeat
38    Al Hilal  4-1-4-1       1  ...   6            6     Win
39    Al-Batin  4-3-2-1       0  ...   1            6  Defeat
40  Al Ettifaq  4-2-3-1       1  ...   4            4     Win
41    Al Fayha    4-4-2       0  ...   0            4  Defeat
42  Al Faisaly  4-2-3-1       1  ...   1            1     Win
43   Ohod Club  4-2-3-1       0  ...   0            1  Defeat
44      Alahli  4-1-4-1       1  ...   4            4     Win
45    Al Hazem  4-2-3-1       0  ...   2            4  Defeat
46    Al Wehda  4-2-3-1       1  ...   2            2     Win
47     Al Raed  4-2-3-1       0  ...   1            2  Defeat
48    Al Fateh  4-2-3-1       1  ...   3            3  Defeat
49    Al Hilal  4-2-3-1       0  ...   9            3     Win

[50 rows x 12 columns]

In [ ]:
season_df.tail(50)

Team1   format  IsHome  ... pts opponent_pts  result
798  Al Ittihad    4-3-3       1  ...  19           19     Win
799   Al Shabab  4-1-4-1       0  ...  28           19  Defeat
800    Al Hazem    3-4-3       1  ...  22           22    Draw
801  Al Ittihad    4-3-3       0  ...  22           22    Draw
802   Al Shabab  4-2-3-1       1  ...  28           28     Win
803    Al Adalh  4-1-4-1       0  ...  14           28  Defeat
804     Al Raed  4-2-3-1       1  ...  31           31    Draw
805    Al Fateh  4-1-4-1       0  ...  18           31    Draw
806    Al Wehda  4-2-3-1       1  ...  33           33     Win
807       Damac  4-2-3-1       0  ...  17           33  Defeat
808      Alahli  4-2-3-1       1  ...  34           34     Win
809    Al Fayha  4-1-4-1       0  ...  27           34  Defeat
810        Abha  4-2-3-1       1  ...  24           24     Win
811  Al Taawoun    3-4-3       0  ...  29           24  Defeat
812  Al Faisaly  4-2-3-1       1  ...  31           31     Win
813    Al Nassr  4-2-3-1       0  ...  42           31  Defeat
814    Al Hilal  4-2-3-1       1  ...  47           47     Win
815  Al Ettifaq  4-2-3-1       0  ...  26           47  Defeat
816       Damac    4-3-3       1  ...  17           17    Draw
817    Al Hilal    4-4-2       0  ...  50           17    Draw
818  Al Ettifaq  4-2-3-1       1  ...  26           26     Win
819   Al Shabab  4-2-3-1       0  ...  31           26  Defeat
820    Al Fayha  4-2-3-1       1  ...  27           27  Defeat
821        Abha  4-2-3-1       0  ...  27           27     Win
822  Al Taawoun    3-4-3       1  ...  29           29     Win
823      Alahli  4-2-3-1       0  ...  37           29  Defeat
824  Al Ittihad  4-2-3-1       1  ...  23           23  Defeat
825    Al Wehda  4-2-3-1       0  ...  36           23     Win
826    Al Nassr  4-1-4-1       1  ...  42           42     Win
827     Al Raed  3-4-1-2       0  ...  32           42  Defeat
828    Al Fateh    4-3-3       1  ...  19           19  Defeat
829    Al Adalh  4-2-3-1       0  ...  14           19     Win
830    Al Hazem    3-4-3       1  ...  23           23    Draw
831  Al Faisaly  4-2-3-1       0  ...  34           23    Draw
832    Al Fateh    4-3-3       1  ...  19           19     Win
833    Al Fayha  4-2-3-1       0  ...  27           19  Defeat
834        Abha  4-2-3-1       1  ...  30           30     Win
835  Al Ittihad  4-3-2-1       0  ...  23           30  Defeat
836     Al Raed    4-4-2       1  ...  32           32     Win
837       Damac    3-5-2       0  ...  18           32  Defeat
838  Al Faisaly  4-2-3-1       1  ...  35           35     Win
839  Al Taawoun  4-2-3-1       0  ...  32           35  Defeat
840      Alahli  4-2-3-1       1  ...  37           37     Win
841    Al Hazem    5-4-1       0  ...  24           37  Defeat
842    Al Wehda  4-2-3-1       1  ...  39           39  Defeat
843   Al Shabab    4-3-3       0  ...  31           39     Win
844    Al Adalh  4-1-4-1       1  ...  17           17  Defeat
845  Al Ettifaq  4-2-3-1       0  ...  29           17     Win
846    Al Nassr  4-1-4-1       1  ...  45           45  Defeat
847    Al Hilal  4-2-3-1       0  ...  51           45     Win

[50 rows x 12 columns]

Adding the round feature so we can visualize over rounds

In [ ]:
df["Round"] = 0

for season in ["2018/2019", "2019/2020"]:

    season_df = df.loc[df["Season"] == season]
    teams = season_df.Team1.unique()
    
    for team in teams:
        df_team = season_df.loc[season_df["Team1"] == team]

        df_team.Round = range(1, len(df_team)+1)

        season_df.loc[season_df["Team1"] == team] = df_team
    
    df.loc[df["Season"] == season] = season_df
        

df.Round = df.Round.astype("int64")

/usr/local/lib/python3.6/dist-packages/pandas/core/generic.py:5303: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[name] = value
/usr/local/lib/python3.6/dist-packages/pandas/core/indexing.py:966: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self.obj[item] = s


In [ ]:
df

Team1   format  IsHome    Opponent  ... pts opponent_pts  result Round
0    Al Ettifaq  4-2-3-1       1     Al Raed  ...   0            0    Draw     1
1       Al Raed  4-2-3-1       0  Al Ettifaq  ...   0            0    Draw     1
2      Al Hazem  4-2-3-1       1    Al Wehda  ...   0            0    Draw     1
3      Al Wehda  4-2-3-1       0    Al Hazem  ...   0            0    Draw     1
4     Ohod Club  4-3-2-1       1    Al Nassr  ...   0            0  Defeat     1
..          ...      ...     ...         ...  ...  ..          ...     ...   ...
843   Al Shabab    4-3-3       0    Al Wehda  ...  31           39     Win    23
844    Al Adalh  4-1-4-1       1  Al Ettifaq  ...  17           17  Defeat    23
845  Al Ettifaq  4-2-3-1       0    Al Adalh  ...  29           17     Win    23
846    Al Nassr  4-1-4-1       1    Al Hilal  ...  45           45  Defeat    23
847    Al Hilal  4-2-3-1       0    Al Nassr  ...  51           45     Win    23

[848 rows x 13 columns]

In [ ]:
not_matching_rounds_index = []
for x in range(0, len(df), 2):
    if df.loc[x, "Round"] != df.loc[x+1, "Round"]:
        not_matching_rounds_index.append(x)
        not_matching_rounds_index.append(x+1)

In [ ]:
df.loc[not_matching_rounds_index]

Team1   format  IsHome    Opponent  ... pts opponent_pts  result Round
90   Al Ettifaq  4-2-3-1       1      Alahli  ...   8            8     Win     5
91       Alahli  4-2-3-1       0  Al Ettifaq  ...  13            8  Defeat     6
92    Al Shabab  4-2-3-1       1    Al Hilal  ...   9            9  Defeat     6
93     Al Hilal  4-1-4-1       0   Al Shabab  ...  12            9     Win     5
96   Al Ettifaq  4-2-3-1       1    Al Hazem  ...  11           11     Win     6
..          ...      ...     ...         ...  ...  ..          ...     ...   ...
685      Alahli  4-2-3-1       0    Al Hilal  ...  23           24  Defeat    13
696  Al Faisaly  4-2-3-1       1     Al Raed  ...  20           20  Defeat    13
697     Al Raed  4-2-3-1       0  Al Faisaly  ...  23           20     Win    14
698    Al Hilal  4-2-3-1       1    Al Wehda  ...  27           27     Win    13
699    Al Wehda  4-2-3-1       0    Al Hilal  ...  24           27  Defeat    14

[96 rows x 13 columns]

In [ ]:
goal_keepers_2018 = {'Al Ettifaq', 'Al Raed', 'Al Hazem', 'Al Wehda', 'Ohod Club',
       'Al Nassr', 'Qadisiyah', 'Al Fateh', 'Al Hilal', 'Al Fayha',
       'Al Shabab', 'Al Ittihad', 'Al Faisaly', 'Al-Batin', 'Alahli',
       'Al Taawoun'}


goal_keepers_2020 = {'Al Nassr', 'Damac', 'Alahli', 'Al Adalh', 'Al Hilal':'abdullah al maiouf', 'Abha',
       'Al Taawoun', 'Al Hazem', 'Al Ittihad', 'Al Raed', 'Al Fateh',
       'Al Shabab', 'Al Faisaly', 'Al Fayha', 'Al Wehda', 'Al Ettifaq'}


In [ ]:
df.head(10)

Team1   format  IsHome    Opponent  ... pts opponent_pts  result Round
0  Al Ettifaq  4-2-3-1       1     Al Raed  ...   0            0    Draw     1
1     Al Raed  4-2-3-1       0  Al Ettifaq  ...   0            0    Draw     1
2    Al Hazem  4-2-3-1       1    Al Wehda  ...   0            0    Draw     1
3    Al Wehda  4-2-3-1       0    Al Hazem  ...   0            0    Draw     1
4   Ohod Club  4-3-2-1       1    Al Nassr  ...   0            0  Defeat     1
5    Al Nassr  4-3-2-1       0   Ohod Club  ...   0            0     Win     1
6   Qadisiyah    4-3-3       1    Al Fateh  ...   0            0    Draw     1
7    Al Fateh  4-2-3-1       0   Qadisiyah  ...   0            0    Draw     1
8    Al Hilal  4-1-3-2       1    Al Fayha  ...   0            0     Win     1
9    Al Fayha    3-5-2       0    Al Hilal  ...   0            0  Defeat     1

[10 rows x 13 columns]